monitor/server.cpp

```diff
@@ -20,146 +20,222 @@
 #include "server.hpp"
 #include "monitor.hpp"
 
+#include <chrono>
+
 #include "library/elliptics.h"
 #include "library/logger.hpp"
 #include "http_miscs.hpp"
 
 namespace ioremap { namespace monitor {
-
-class handler: public std::enable_shared_from_this<handler> {
+class handler : public std::enable_shared_from_this<handler> {
 public:
-	handler(monitor &mon, boost::asio::io_service &io_service)
+	handler(monitor &mon, boost::asio::ip::tcp::socket socket)
 	: m_monitor(mon)
-	, m_socket(io_service)
-	, m_remote("")
-	{}
+	, m_socket(std::move(socket))
+	, m_remote([this]() {
+		const auto &endpoint = m_socket.remote_endpoint();
+		std::ostringstream stream;
+		stream << endpoint.address().to_string() << ":" << endpoint.port();
```

Comment:
It wraps IPv6 address with `[]` which is not needed here and useful in full url. I've added `to_string` function which can do both.